components/encryption/src/file_dict_file.rs

```diff
@@ -172,21 +173,28 @@ impl FileDictionaryFile {
                 // parse the remained records
                 while !remained.is_empty() {
                     // If the parsing get errors, manual intervention is required to recover.
-                    let (used_size, file_name, mode) = Self::parse_next_record(remained)?;
-                    remained.consume(used_size);
-                    match mode {
-                        LogRecord::INSERT(info) => {
-                            file_dict.files.insert(file_name, info);
-                        }
-                        LogRecord::REMOVE => {
-                            let original = file_dict.files.remove(&file_name);
-                            if original.is_none() {
-                                return Err(box_err!(
-                                    "Try to recovery from log file but remove a null entry, file name: {}",
-                                    file_name
-                                ));
+                    match Self::parse_next_record(remained) {
+                        Ok((used_size, file_name, mode)) => {
+                            remained.consume(used_size);
+                            match mode {
+                                LogRecord::INSERT(info) => {
+                                    file_dict.files.insert(file_name, info);
+                                }
+                                LogRecord::REMOVE => {
+                                    let original = file_dict.files.remove(&file_name);
+                                    if original.is_none() {
+                                        return Err(box_err!(
+                                            "Try to recovery from log file but remove a null entry, file name: {}",
+                                            file_name
+                                        ));
+                                    }
+                                }
                             }
                         }
+                        Err(Error::ParseIncomplete) => {
```

Comment:
upon tail record corruption, can we print some info regarding the last complete record, so we can debug what's being truncated?

Reply:
maybe just warn info is enough, what is the use of returning debug information like `TailRecordParseIncomplete(discarded bytes)`?

Reply:
From the info log I want to know for sure which record is discarded. Say in the info log of the key manager, we know that the last two encrypted files created are 001.sst and 002.sst. If we can print an info log here saying the last **complete** record was for 001.sst, we are sure 002.sst is the one being dropped when debugging.

Reply:
I will attach the info for the last complete record with the error type

Reply:
Just log before return break while loop, because `parse_next_record` only operate the remained bytes but not the parsed parts